HLTrigger/Configuration/python/customizeHLTTrackingForPhaseI2017.py

```diff
@@ -281,49 +427,89 @@ def _copy(old, new, skip=[]):
+	        lockHits = cms.bool(True),
+	        TTRHBuilder = cms.string('hltESPTTRHBWithTrackAngle'),
+	        foundHitBonus = cms.double(5.0),
+	        updator = cms.string('hltESPKFUpdator'),
+	        alwaysUseInvalidHits = cms.bool(False),
+	        requireSeedHitsInRebuild = cms.bool(True),
+	        keepOriginalIfRebuildFails = cms.bool(False),
+	        estimator = cms.string('hltESPChi2ChargeMeasurementEstimator16'),
+	        propagatorOpposite = cms.string('PropagatorWithMaterialParabolicMfOpposite'),
+	        minNrOfHitsForRebuild = cms.int32(5),
+	        maxDPhiForLooperReconstruction = cms.double(2.0),
+	        maxPtForLooperReconstruction = cms.double(0.7),
+	        cleanTrajectoryAfterInOut = cms.bool( False ),
+	        useHitsSplitting = cms.bool( False ),
+	        doSeedingRegionRebuilding = cms.bool( False )
+    )	
+
+    process.hltIter2PFlowCkfTrackCandidates.TrajectoryBuilder = cms.string('HLTIter2GroupedCkfTrajectoryBuilderIT')
+    process.hltIter2PFlowCkfTrackCandidates.TrajectoryBuilderPSet = cms.PSet(refToPSet_ = cms.string('HLTIter2GroupedCkfTrajectoryBuilderIT'))
+
+
+
+    process.HLTIterativeTrackingIteration2 = cms.Sequence( process.hltIter2ClustersRefRemoval + process.hltIter2MaskedMeasurementTrackerEvent + process.hltIter2PixelLayerTriplets + process.hltIter2PFlowPixelTrackingRegions + process.hltIter2PFlowPixelClusterCheck + process.hltIter2PFlowPixelHitDoublets + process.hltIter2PFlowPixelHitTriplets + process.hltIter2PFlowPixelSeeds + process.hltIter2PFlowCkfTrackCandidates + process.hltIter2PFlowCtfWithMaterialTracks + process.hltIter2PFlowTrackCutClassifier + process.hltIter2PFlowTrackSelectionHighPurity )
+
+    # Need to operate on Paths as well...
+    for seqs in [process.sequences_(), process.paths_()]:
+        for seqName, seq in seqs.iteritems():
+            from FWCore.ParameterSet.SequenceTypes import ModuleNodeVisitor
+            l = list()
+            v = ModuleNodeVisitor(l)
+            seq.visit(v)
+
+            if process.hltPixelTracks in l and not process.hltPixelLayerQuadruplets in l:
+                seq.remove(process.hltPixelLayerTriplets) # note that this module does not necessarily exist in sequence 'seq', if it doesn't, it does not get removed
+                index = seq.index(process.hltPixelTracksHitDoublets)
+                seq.insert(index,process.hltPixelLayerQuadruplets)
+                index = seq.index(process.hltPixelTracksHitTriplets)
+                seq.remove(process.hltPixelTracksHitTriplets)
+                seq.insert(index, process.hltPixelTracksHitQuadruplets)
 
     # Remove entirely to avoid warning from the early deleter
     del process.hltPixelTracksHitTriplets
 
-    return process
+    for producer in producers_by_type(process,"PixelTripletHLTEDProducer"):
+        if "hltIter1PFlowPixelHitTriplets" in producer.label():
+
+    	   	for seqs in [process.sequences_(), process.paths_()]:
```

Comment:
Could the logic be included in the previous loop over all sequences, paths, and endpaths? (maybe collect beforehand the set of `PixelTripletHLTEDProducer`'s and do the necessary replacements for them.